### DISCUSSION TITLE
How to specify multiple colors + opaque?

### DISCUSSION THREAD

**Discussion by b1zar** [ImageMagick/ImageMagick#2390] [Help] [answered]
Hey. Tell me how to fill specific colors in one line? Let's say I need to know the number of pixels of 3 colors:
1)rgb(241,241,241)
2)rgb(242,242,242)
3)rgb(243,243,243)

I am using the command to get one color "magick convert 2.png -fill black +opaque "rgb(240,240,240)" -format %%c histogram:info:> Magic.txt"

how to specify several colors here? And will it be possible to draw the output of the histogram with the average gray value?

version: ImageMagick-7.0.10-24-Q16-HDRI-x64-dll, Windows 10

**Comment by snibgo** [comment, marked as answer]
I'm unsure what you are asking.

The histogram will give you a count of every colour. A shell script could pick out just the colours of interest.

> how to specify several colors here?

IM commands can be long, so you could change all the colours except those three to black. How? Assuming no pixels are already transparent: make the first colour transparent ("-transparent rgb(241,241,241)"), then the second and third colours. Then flip transparency ("-channel a -negate +channel"). Then flatten against black ("-background black -layers flatten").

> And will it be possible to draw the output of the histogram with the average gray value?

You can convert the image to grayscale, and draw the histogram.

**Reply by b1zar** [reply]
This is what you need. I went around transparency but didn't think to apply it this way)) is it possible to display transparency in the histogram without using fill?

**Comment by snibgo** [comment]
>  is it possible to display transparency in the histogram without using fill?

The histogram graph shows only the RGB channels. "-alpha extract" makes a grayscale version of the image from the alpha channel, so you can make a histogram graph from that.

If you are adventurous you could combine the two histogram graphs into a single graph.
